src/main/java/com/sourcegraph/telemetry/GraphQlLogger.java

```diff
@@ -65,9 +65,13 @@ private static JsonObject addGlobalEventParameters(
   // This could be exposed later (as public), but currently, we don't use it externally.
   private static CompletableFuture<Boolean> logEvent(
       @NotNull Project project, @NotNull Event event) {
-    return CodyAgentService.getAgent(project)
-        .thenApply(agent -> agent.getServer().logEvent(event))
-        .thenApply((ignored) -> true)
-        .exceptionally((ignored) -> false);
+    return CodyAgentService.withAgent(project)
+        .thenCompose(
+            agent ->
```

Comment:
nit: you can just use `it` instead of the named argument

Reply:
It is java :)

Reply:
ah...